starter-code/HeterogeneousRecyclerView/app/src/main/java/com/charlesdrews/heterogeneousrecyclerview/FootballRecyclerViewAdapter.java

```diff
@@ -17,6 +18,48 @@
 
 public class FootballRecyclerViewAdapter extends RecyclerView.Adapter {
 
+    private List<BaseFootballObject> mBaseFootballObjects;
+
+    public FootballRecyclerViewAdapter(List<BaseFootballObject> listOfFutbol) {
+        mBaseFootballObjects = listOfFutbol;
+    }
+
+    @Override
+    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        if (viewType == 0){
+            View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.team_list_entry, null);
+            return new TeamViewHolder(view);
+        } else{
+            View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.player_list_entry, null);
+            return new PlayerViewHolder(view);
+        }
+
+    }
+
+    @Override
+    public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
+        BaseFootballObject baseFootballObject = mBaseFootballObjects.get(holder.getAdapterPosition());
+        if (Team.class.equals(baseFootballObject.getClass())){
+            ((TeamViewHolder) holder).bindDataToViews((Team)mBaseFootballObjects.get(holder.getAdapterPosition()));
+        }else{
+            ((PlayerViewHolder) holder).bindDataToViews((Player)mBaseFootballObjects.get(holder.getAdapterPosition()));
+        }
+    }
+
+    @Override
+    public int getItemViewType(int position) {
+        if (Team.class.equals(mBaseFootballObjects.get(position).getClass())){
+            return 0;
+        } else {
+            return 1;
```

Comment:
I recommend using int constants here, like `TEAM` and `PLAYER` to make your code more readable.